pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/loader/SegmentPreProcessor.java

```diff
@@ -62,6 +63,16 @@ public void process() throws Exception {
       return;
     }
 
+    File[] directoryListing = _indexDir.listFiles();
```

Comment:
Please add comments here to explain why we are doing this. In the future, if other people look at this piece of code, they won't get confused.